port/cpl_vsil_libarchive.cpp

```diff
@@ -529,4 +576,27 @@ void VSIInstall7zFileHandler(void)
         "/vsi7z/", new VSILibArchiveFilesystemHandler("/vsi7z"));
 }
 
+#ifdef HAVE_BLAKE2
```

Comment:
if the #ifdef HAVE_BLAKE2 is really needed, it should be moved inside the function body, so that the function is always defined in the HAVE_LIBARCHIVE branch

Reply:
Thanks. I was trying to follow the pattern of the #ifdefs for `VSIInstall7zFileHandler` but will change it to a single function body instead.

Reply:
I wasn't perhaps clear. You need 2 bodies. One at the top of the file in the #ifndef HAVE_LIBARCHIVE branch (without a #ifdef HAVE_BLAKE2) condition.
And the one at the bottom of the file, which is now fine with your modifications

Reply:
Thanks. I hope this is correct now, with everything controlled just by the `#ifndef HAVE_LIBARCHIVE`.